content/post/2022-08-06-netflix-description-analysis/index.Rmd

```diff
@@ -0,0 +1,174 @@
+
+-   `tidyverse` for the data manipulation
+
+-   `tidymodels` for the modelling
+
+-   `tidytext` for the analysis of the text data
+
+-   `textrecipes` for additional recipes step for text data
+
+## Importing the pacakges
+
+```{r message=FALSE, warning=FALSE}
+library(tidyverse)
+library(tidymodels)
+library(tidytext)
+library(textrecipes)
+library(hrbrthemes)
+theme_set(theme_light(base_size = 20))
+```
+
+## Read data 
+
+```{r include=FALSE}
+path <- "/Users/ieo5571/Documents/blog_data_science_bioinfo/Data/tuesdata.csv"
+```
+
+The data is in *.csv* in order to read it we can use the `readr::read_csv()` function. Once the data are read and saved in the `netflix_titles` variable.
+
+```{r}
+netflix_titles <- readr::read_csv(path)
+
+```
+
+In order to have a look to the data we can use a very handy function from the `dplyr` which is `glimpse()`.
+
+```{r}
+dplyr::glimpse(netflix_titles)
+```
+
+We can see that we have 7,787 rows and 12 columns. It seams also that the dataset contains also *TV Shows.* We can filter them out and select only the following columns:
+
+-   `title`
+
+-   `description`
+
+-   `listed_in`
+
+Moreover using `tidyr::separate_rows()` we can split the word of the `description` column and create a table with a row for each word.
+
+```{r}
+netflix_films <- 
+  netflix_titles %>%  
+  filter(tolower(type)  == "movie") %>% 
+  select(title, description, listed_in) %>% 
+  separate_rows(listed_in, sep = ", ")
+```
+
+## EDA
+
+Let's first of all perform some **Exploratory Data Analysis** (**EDA**). A useful information could be counting the number of films by category. This can be done as follows:
+
+```{r}
+netflix_films %>% 
+  count(listed_in, sort = T) %>% 
+  mutate(listed_in = fct_reorder(listed_in, n)) %>% 
+  
+  ggplot(aes(x = n, y = listed_in, fill = n)) +
+  geom_segment(aes(xend = 0, yend = listed_in), 
+               color = "grey50") +
+  geom_point(aes(size = abs(n)), 
+             shape = 21, 
+             color = "grey50", 
+             show.legend = FALSE)  +
+  scale_fill_viridis_c(option = "C")  +
+  scale_size_continuous(range = c(4, 7)) +
+  labs(x = "Number of films",  y = NULL)
+```
+
+We have lots of categories, some with lots of observation other with very few. This is not the optimal condition for a *machine learning model*. We can follow different path here. We can select some categories and collapse all the other into a macro-category or we can select just few of them and drop all the others. For the sake of this simple tutorial we will select just some categories. We are going to select:
+
+-   Horror Movies
+
+-   Children & Family Movies
+
+-   Documentaries
+
+We are left with 1,630.
+
+```{r}
+category_to_select <- 
+    c("Horror Movies", "Children & Family Movies",
+      "Documentaries")
+
+netflix_films <- netflix_films %>%  
+  filter(listed_in %in% category_to_select)
```

Comment:
è necessario listed in ?

Reply:
Nel codice  intendi? Nel codice listed_in è la colonna che contiene le categorie per filtrarlo si che serve